java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/map/TieredStoreMapTest.java

```diff
@@ -0,0 +1,191 @@
+import com.hazelcast.memory.Capacity;
+import com.hazelcast.simulator.hz.HazelcastTest;
+import com.hazelcast.simulator.test.BaseThreadState;
+import com.hazelcast.simulator.test.annotations.Prepare;
+import com.hazelcast.simulator.test.annotations.Setup;
+import com.hazelcast.simulator.test.annotations.Teardown;
+import com.hazelcast.simulator.test.annotations.TimeStep;
+import com.hazelcast.simulator.test.annotations.Verify;
+import com.hazelcast.simulator.tests.helpers.KeyLocality;
+import com.hazelcast.simulator.tests.map.helpers.tasks.ClearTsDirectoryTask;
+import com.hazelcast.simulator.tests.map.helpers.tasks.GetHybridLogLengthTask;
+import com.hazelcast.simulator.tests.map.helpers.tasks.GetMapConfigTask;
+import com.hazelcast.simulator.worker.loadsupport.Streamer;
+import com.hazelcast.simulator.worker.loadsupport.StreamerFactory;
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.Assert;
+
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+
+import static com.hazelcast.jet.core.test.JetAssert.assertTrue;
+import static com.hazelcast.memory.MemoryUnit.BYTES;
+import static com.hazelcast.simulator.tests.helpers.KeyLocality.SHARED;
+import static com.hazelcast.simulator.tests.helpers.KeyUtils.generateIntKeys;
+import static com.hazelcast.simulator.utils.GeneratorUtils.generateByteArray;
+
+/**
+ * This test is running as part of release verification simulator test. Hence every change in this class should be
+ * discussed with QE team since it can affect release verification tests.
+ */
+public class TieredStoreMapTest extends HazelcastTest {
+    // properties
+    public int keyDomain = 1_500_000;
+    public int minValueByteArrayLength = 1;
+    public int maxValueByteArrayLength = 150_000;
+    public boolean clearTsDirectoryOnPrepare = true;
+    public boolean fillOnPrepare = true;
+    public boolean destroyOnExit = true;
+
+    public KeyLocality keyLocality = SHARED;
+
+    private IMap<Integer, byte[]> map;
+    private int[] keys;
+    private IExecutorService executor;
+
+    @Setup
+    public void setUp() throws ReflectiveOperationException {
+        keys = generateIntKeys(keyDomain, keyLocality, targetInstance);
+        executor = targetInstance.getExecutorService(name);
+        map = targetInstance.getMap(name);
+        Assert.assertTrue("Disk Tier Config must be enabled for map: " + name, isTsEnabledForMap());
+    }
+
+    @Prepare(global = true)
+    public void prepare() {
+        if (clearTsDirectoryOnPrepare) {
+            clearTsDirectory();
+        }
+        if (fillOnPrepare) {
+            fillMap();
+        }
+    }
+
+    @Verify
+    public void verify() {
+        Capacity hybridLogLength = Capacity.of(getHybridLogLengthSum(), BYTES);
+        logger.info("Sum of hybrid log length from all members: " + hybridLogLength.toPrettyString());
+        assertTrue("Hybrid log is equal to 0. Tiered Storage was not used.", hybridLogLength.bytes() > 0);
+    }
```

Comment:
Are these kind of tests just "stability" tests? With this test we pass even if we store/retrieve invalid or inconsistent data.

Reply:
Yes, this test will be run for 48h and we are just checking if it is stable on cluster which will have also other tests running at the same time.

Reply:
...but if you think that is worth to add something extra without complex logic, then it can be added.

Reply:
If this is what other tests are doing, I'm okay with this. I don't know the capabilities of our simulator validation tests, but if we can calculate the values as a function of the key, we can iterate over all entries in `verify`.

Reply:
Yes, the other tests are similar - very simple, sometimes without real validation after. 
Possibility to calculate values will require partial de-randomization and we cannot store details due to amount of data.
I think would be easier and safer to keep this as it is. If something will be wrong with a client or the cluster - we will know that.